code/modules/mob/living/simple_animal/bot_swarm/swarmer.dm

```diff
@@ -0,0 +1,598 @@
+
+/mob/living/simple_animal/hostile/swarmer/proc/Fabricate(var/atom/fabrication_object,var/fabrication_cost = 0)
+	if(!isturf(loc))
+		src << "<span class='warning'>This is not a suitable location for fabrication. We need more space.</span>"
+	if(resources >= fabrication_cost)
+		resources -= fabrication_cost
+	else
+		src << "<span class='warning'>You do not have the necessary resources to fabricate this object.</span>"
+		return 0
+	new fabrication_object(loc)
+	return 1
+
+/mob/living/simple_animal/hostile/swarmer/proc/Integrate(var/obj/item/target)
+	if(resources >= 100)
+		src << "<span class='warning'>We cannot hold more materials!</span>"
+		return
+   		//Check if any entries are either MAT_METAL or MAT_GLASS
+	if((MAT_METAL in target.materials) || (MAT_GLASS in target.materials))
+		resources++
+		do_attack_animation(target)
+		changeNext_move(CLICK_CD_MELEE)
+		var/obj/effect/swarmer/integrate/I = new /obj/effect/swarmer/integrate(get_turf(target))
+		I.pixel_x = target.pixel_x
+		I.pixel_y = target.pixel_y
+		I.pixel_z = target.pixel_z
+		if(istype(target, /obj/item/stack))
+			var/obj/item/stack/S = target
+			S.use(1)
+			if(S.amount)
+				return
+		qdel(target)
+	else
+		src << "<span class='warning'>\the [target] is incompatible with our internal matter recycler.</span>"
+		return
+
+/mob/living/simple_animal/hostile/swarmer/proc/DisIntegrate(var/atom/movable/target)
+	new /obj/effect/swarmer/disintegration(get_turf(target))
+	do_attack_animation(target)
+	changeNext_move(CLICK_CD_MELEE)
+	target.ex_act(3)
+	return
+
+/mob/living/simple_animal/hostile/swarmer/proc/DisperseTarget(var/mob/living/target)
+	if(target != src)
+		src << "<span class='info'>Attempting to remove this being from our presence.</span>"
+		if(src.z != ZLEVEL_STATION)
+			src << "<span class='warning'>Our bluespace transceiver cannot locate a viable bluespace link, our teleportation abilities are useless in this area.</span>"
+			return
+		if(do_mob(src, target, 30))
+			var/cycle
+			for(cycle=0,cycle<100,cycle++)
+				var/random_location = locate(rand(37,202),rand(75,192),ZLEVEL_STATION)//Drunk dial a turf in the general ballpark of the station
+				if(istype(random_location, /turf/simulated/floor))
+					var/turf/simulated/floor/F = random_location
+					if(F.air)
+						var/datum/gas_mixture/A = F.air
+						if(A.oxygen >= 16 && !A.toxins && A.carbon_dioxide < 10 && !A.trace_gases.len)//Can most things breathe in this location?
+							if((A.temperature > 270) && (A.temperature < 360))//Not too hot, not too cold
+								var/pressure = A.return_pressure()
+								if((pressure > 20) && (pressure < 550))//Account for crushing pressure or vaccuums
+									if(ishuman(target))//If we're getting rid of a human, slap some zipties on them to keep them away from us a little longer
+										var/obj/item/weapon/restraints/handcuffs/cable/zipties/Z = new /obj/item/weapon/restraints/handcuffs/cable/zipties(src)
+										var/mob/living/carbon/human/H = target
```

Comment:
assuming `apply_cuffs()` is sane and doesn't just run on the assumption that it's being passed a human this typecasting should be unnecessary